### PR TITLE
[docs-only] Document setting up the profiler

### PR DESCRIPTION
**Pull request by refs**

### CODE REVIEW

File: docs/ocis/development/profiling.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+The previous patch will:
+
+1. import `net/http/pprof`, which will register debug handlers in `DefaultServeMux`.
+2. define a `init()` function that starts an http server with the previously registered handlers.
+
+With everything running one should have access to http://localhost:1234/debug/pprof/
+
+## 3. Replace ocis go.mod to local reva and build a new binary
```
Comment (pascalwengerter) [inline docs/ocis/development/profiling.md:58]:
```suggestion
## 3. Replace reva in oCIS go.mod with local version and build a new binary
```

File: docs/ocis/development/profiling.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+```console
+$ go get -u github.com/google/pprof
+```
+
+### Collecting samples
+
+Collect 30 seconds of samples:
+
+```console
+$ prof -web http://:1234/debug/pprof/profile\?seconds\=30
```
Comment (pascalwengerter) [inline docs/ocis/development/profiling.md:109]:
```suggestion
$ pprof -web http://:1234/debug/pprof/profile\?seconds\=30
```

File: docs/ocis/development/profiling.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+Once the collection is done a browser tab will open with the result `svg`, looking similar to this:
+
+![img](https://i.imgur.com/vo0EbcX.jpg)
+
+For references on how to interpret this graph, [continue reading here](https://github.com/google/pprof/blob/master/doc/README.md#interpreting-the-callgraph).
+
+## References
+
+- https://medium.com/swlh/go-profile-your-code-like-a-master-1505be38fdba
+- https://dave.cheney.net/
```
Comment (pascalwengerter) [inline docs/ocis/development/profiling.md:121]: Do you want to recommend the blog as is, or a specific post, e.g. https://dave.cheney.net/2013/07/07/introducing-profile-super-simple-profiling-for-go-programs?

File: docs/ocis/development/profiling.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+ func readPropfind(r io.Reader) (pf propfindXML, status int, err error) {
+        c := countingReader{r: r}
+```
+
+The previous patch will:
+
+1. import `net/http/pprof`, which will register debug handlers in `DefaultServeMux`.
+2. define a `init()` function that starts an http server with the previously registered handlers.
+
+With everything running one should have access to http://localhost:1234/debug/pprof/
```
Comment (pascalwengerter) [inline docs/ocis/development/profiling.md:56]: I'd mention following the setup/build guide within Reva since you need the binary for oCIS?

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Minor nit-picks. Very nice read, running it kinda failed me with `pprof: failed to execute dot. Is Graphviz installed? Error: exec: "dot": executable file not found in $PATH
` (but there's something off with my Golang setup I suppose since I needed to start it via `./go/bin/pprof -web http://:1234/debug/pprof/profile\?seconds\=30` instead of your `$ pprof ...` cmd)

File: docs/ocis/development/profiling.md (hunk 3 above)
Comment (refs) [reply docs/ocis/development/profiling.md:121]: aah, sweet sweet catch. I meant to provide with the article. Thanks

**Comment by refs** [comment]
@pascalwengerter 

It just so happened I had `dot` on my `$PATH`. Having a look at google/pprof:

> pprof reads a collection of profiling samples in profile.proto format and generates reports to visualize and help analyze the data. It can generate both text and graphical reports (through the use of the dot visualization package).

[_sauce_](https://github.com/google/pprof/tree/f964ff60559510aa6b3398b59d6d19c42da2ea37)

Instead I will rephrase to read google/pprof installation guide 👍

File: docs/ocis/development/profiling.md (hunk 4 above)
Comment (refs) [reply docs/ocis/development/profiling.md:56]: Hm, how within Reva? You mean linking [here](https://github.com/cs3org/reva/#build-and-run-it-yourself)?

Comment (pascalwengerter) [reply docs/ocis/development/profiling.md:65]: Exactly. Probably it's obvious for most people but I'd be lost without at least some knowledge of how Golang/reva+oCIS work & play together

**Review by dragotin — CHANGES_REQUESTED** [review]
Tbh, I think it is very brief. 

I love to understand why I should do things. For example
- What is Reva, I thought I debug oCIS? 
- Why do I need a special patch in Reva to get the traces?
- Why do I have to edit the go.mod?
- What are the env vars OCIS_LOG_PRETTY etc.?
- Is `pprof` the welknown super virus "Power Professor" that steals my computers electrical energy to farm coins?
-  etc.

I would appreciate appreciate a bit more "background information", answering questions like the ones above.

**Comment by micbar** [comment]
@dragotin @refs please update or close

**Comment by refs** [comment]
> @dragotin @refs please update or close

I'll update it

**Comment by refs** [comment]
> Tbh, I think it is very brief.
> 
> I love to understand why I should do things. For example
> 
>     * What is Reva, I thought I debug oCIS?
> 
>     * Why do I need a special patch in Reva to get the traces?
> 
>     * Why do I have to edit the go.mod?
> 
>     * What are the env vars OCIS_LOG_PRETTY etc.?
> 
>     * Is `pprof` the welknown super virus "Power Professor" that steals my computers electrical energy to farm coins?
> 
>     * etc.
> 
> 
> I would appreciate appreciate a bit more "background information", answering questions like the ones above.

Took me a long one to get back to this one, but I think I addressed all your questions with the latest commit :)

**Review by micbar — APPROVED** [review]

**Pull request merged**